test/init.test.ts

```diff
@@ -67,22 +71,66 @@ describe('Flagsmith.init', () => {
+        mockFetch.mockResolvedValueOnce({status: 404, text: async () => ''})
         await expect(flagsmith.init(initConfig)).rejects.toThrow(Error);
     });
     test('identifying with new identity should not carry over previous traits for different identity', async () => {
         const onChange = jest.fn()
-        const now = Date.now();
-        const identityA = `test_identity_a_${now}`
-        const identityB = `test_identity_b_${now}`
-        const {flagsmith,initConfig} = getFlagsmith({onChange, identity:identityA, traits: {a:`example`}})
+        const identityA = `test_identity_a`
+        const identityB = `test_identity_b`
+        const {flagsmith,initConfig,mockFetch} = getFlagsmith({onChange, identity:identityA, traits: {a:`example`}})
+        mockFetch.mockResolvedValueOnce({status: 200, text: () => fs.readFile(`./test/data/identities_${identityA}.json`, 'utf8')})
         await flagsmith.init(initConfig);
         expect(flagsmith.getTrait("a")).toEqual(`example`)
+        mockFetch.mockResolvedValueOnce({status: 200, text: () => fs.readFile(`./test/data/identities_${identityB}.json`, 'utf8')})
         await flagsmith.identify(identityB)
         expect(flagsmith.getTrait("a")).toEqual(undefined)
+        mockFetch.mockResolvedValueOnce({status: 200, text: () => fs.readFile(`./test/data/identities_${identityA}.json`, 'utf8')})
         await flagsmith.identify(identityA)
         expect(flagsmith.getTrait("a")).toEqual(`example`)
+        mockFetch.mockResolvedValueOnce({status: 200, text: () => fs.readFile(`./test/data/identities_${identityB}.json`, 'utf8')})
         await flagsmith.identify(identityB)
         expect(flagsmith.getTrait("a")).toEqual(undefined)
     });
+    test('identifying with transient identity should request the API correctly', async () => {
+        const onChange = jest.fn()
+        const testTransientIdentity = `test_transient_identity`
+        const {flagsmith,initConfig,mockFetch} = getFlagsmith({onChange, identity: {identifier: testTransientIdentity, transient: true}})
+        mockFetch.mockResolvedValueOnce({status: 200, text: () => fs.readFile(`./test/data/identities_${testTransientIdentity}.json`, 'utf8')})
+        await flagsmith.init(initConfig);
+        expect(mockFetch).toHaveBeenCalledWith(`https://edge.api.flagsmith.com/api/v1/identities/?identifier=${testTransientIdentity}&transient=true`,
+            expect.objectContaining({method: 'GET'}),
+        )
+    });
+    test('identifying with transient traits should request the API correctly', async () => {
+        const onChange = jest.fn()
+        const testIdentityWithTransientTraits = `test_identity_with_transient_traits`
+        const {flagsmith,initConfig,mockFetch} = getFlagsmith({onChange, identity: testIdentityWithTransientTraits,
+            traits: {number_trait:1, string_trait:"Example",transient_trait:{transient:true,value:"Example"}}
+        })
```

Comment:
I expected an optional param for a list of transient traits to maintain backward compatibility and to have better usability/UX. I would expect to have this usage of the SDK:
```
const {flagsmith,initConfig,mockFetch} = getFlagsmith({onChange, identity: testIdentityWithTransientTraits,
   traits: {number_trait:1, string_trait:"Example"},
   transient_traits: {transient_trait_1: "Example", transient_trait_2: 3}
        })

```

Reply:
As discussed (results [here](https://github.com/Flagsmith/flagsmith/issues/4278#issuecomment-2248140003)), we're going forward with the current implementation.